test/cppsuite/tests/cursor_bound_02.cpp

```diff
@@ -0,0 +1,313 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "src/bound/bound_set.h"
+#include "src/common/constants.h"
+#include "src/common/random_generator.h"
+#include "src/main/test.h"
+
+using namespace test_harness;
+
+/*
+ * In this test, we want to verify that search_near with prefix enabled only traverses the portion
+ * of the tree that follows the prefix portion of the search key. The test is composed of a populate
+ * phase followed by a read phase. The populate phase will insert a set of random generated keys
+ * with a prefix of aaa -> zzz. During the read phase, we have one read thread that performs:
+ *  - Spawning multiple threads to perform one prefix search near.
+ *  - Waiting on all threads to finish.
+ *  - Using WiredTiger statistics to validate that the number of entries traversed is within
+ * bounds of the search key.
+ */
+class cursor_bound_02 : public test {
+    uint64_t keys_per_prefix = 0;
+    uint64_t srchkey_len = 0;
```

Comment:
`srchkey_len` -> `search_key_len` looks better? If you proceed, it appears at multiple places in this file.

Reply:
I'm going to leave it as is as it was copied from the original implementation, once we deprecate prefix search we will go through and do a tidying pass of the cursor bound tests.